File: unittests/pytests/friction/TestSlipWeakeningTime.py
#!/usr/bin/env python
#
# ======================================================================
#
# Brad T. Aagaard, U.S. Geological Survey
# Charles A. Williams, GNS Science
# Matthew G. Knepley, University of Chicago
#
# This code was developed as part of the Computational Infrastructure
# for Geodynamics (http://geodynamics.org).
#
# Copyright (c) 2010-2015 University of California, Davis
#
# See COPYING for license information.
#
# ======================================================================
#

## @file unittests/pytests/friction/TestSlipWeakeningTime.py

## @brief Unit testing of SlipWeakeningTime object.

import unittest

from pylith.friction.SlipWeakeningTime import SlipWeakeningTime

# ----------------------------------------------------------------------
class TestSlipWeakeningTime(unittest.TestCase):
  """
  Unit testing of SlipWeakeningTime object.
  """

  def setUp(self):
    """
    Setup test subject.
    """
    self.friction = SlipWeakeningTime()
    return
  

  def test_constructor(self):
    """
    Test constructor.
    """
    return


  def test_factory(self):
    """
    Test factory method.
    """
    from pylith.friction.SlipWeakeningTime import friction_model
    m = friction_model()
    return


# End of file 

File: unittests/libtests/friction/TestSlipWeakeningTime.cc
// -*- C++ -*-
//
// ----------------------------------------------------------------------
//
// Brad T. Aagaard, U.S. Geological Survey
// Charles A. Williams, GNS Science
// Matthew G. Knepley, University of Chicago
//
// This code was developed as part of the Computational Infrastructure
// for Geodynamics (http://geodynamics.org).
//
// Copyright (c) 2010-2015 University of California, Davis
//
// See COPYING for license information.
//
// ----------------------------------------------------------------------
//

#include <portinfo>

#include "TestSlipWeakeningTime.hh" // Implementation of class methods

#include "data/SlipWeakeningTimeData.hh" // USES SlipWeakeningTimeData

#include "pylith/friction/SlipWeakeningTime.hh" // USES SlipWeakeningTime

// ----------------------------------------------------------------------
CPPUNIT_TEST_SUITE_REGISTRATION( pylith::friction::TestSlipWeakeningTime );

// ----------------------------------------------------------------------
// Setup testing data.
void
pylith::friction::TestSlipWeakeningTime::setUp(void)
{ // setUp
  _friction = new SlipWeakeningTime();
  _data = new SlipWeakeningTimeData();
  setupNormalizer();
} // setUp

// ----------------------------------------------------------------------
// Test properties metadata.
void
pylith::friction::TestSlipWeakeningTime::testPropertiesMetadata(void)
{ // testPropertiesMetadata
  SlipWeakeningTime model;

  CPPUNIT_ASSERT_EQUAL(5, model._metadata.numDBProperties());
  const char* const* names = model._metadata.dbProperties();
  CPPUNIT_ASSERT_EQUAL(std::string("static-coefficient"), 
		       std::string(names[0]));
  CPPUNIT_ASSERT_EQUAL(std::string("dynamic-coefficient"), 
		       std::string(names[1]));
  CPPUNIT_ASSERT_EQUAL(std::string("slip-weakening-parameter"), 
		       std::string(names[2]));
  CPPUNIT_ASSERT_EQUAL(std::string("cohesion"),
		       std::string(names[3]));
  CPPUNIT_ASSERT_EQUAL(std::string("weakening-time"),
		       std::string(names[4]));
} // testPropertiesMetadata

// ----------------------------------------------------------------------
// Test state variable metadata.
void
pylith::friction::TestSlipWeakeningTime::testStateVarsMetadata(void)
{ // testStateVarsMetadata
  SlipWeakeningTime model;

  CPPUNIT_ASSERT_EQUAL(2, model._metadata.numDBStateVars());
  const char* const* names = model._metadata.dbStateVars();
  CPPUNIT_ASSERT_EQUAL(std::string("cumulative-slip"), 
		       std::string(names[0]));
  CPPUNIT_ASSERT_EQUAL(std::string("previous-slip"), 
		       std::string(names[1]));
} // testStateVarsMetadata

// ----------------------------------------------------------------------
// Test hasPropStateVar().
void
pylith::friction::TestSlipWeakeningTime::testHasPropStateVar(void)
{ // testHasPropStateVar
  SlipWeakeningTime material;

  CPPUNIT_ASSERT(material.hasPropStateVar("static_coefficient"));
  CPPUNIT_ASSERT(material.hasPropStateVar("dynamic_coefficient"));
  CPPUNIT_ASSERT(material.hasPropStateVar("slip_weakening_parameter"));
  CPPUNIT_ASSERT(material.hasPropStateVar("cohesion"));
  CPPUNIT_ASSERT(material.hasPropStateVar("weakening_time"));
  CPPUNIT_ASSERT(!material.hasPropStateVar("aaa"));
  CPPUNIT_ASSERT(material.hasPropStateVar("cumulative_slip"));
  CPPUNIT_ASSERT(material.hasPropStateVar("previous_slip"));
} // testHasPropStateVar


// End of file 
